src/controllers/travelControllers.js

```diff
@@ -105,11 +111,40 @@ export const approveTravelRequest = async (req, res) => {
   }
 };
 
-export const mostTravelledDest = async (req, res) => {
-    try {
-      const travelled = await mostTraveled();
-      successResponseWithData(res, statusCode.success, message.oneWayTripCreated, travelled);
-    } catch (error) {
-      errorResponse(res, statusCode.serverError, error);
+const userCanEditOpenRequest = async (req, res) => {
+  const { id } = req.body;
+
+  const userId = req.userData.id;
+
+  try {
+    const result = await checkApprovalStatus(id, userId);
+
+    if (result[0].approval_status !== 'pending') {
+      errorResponse(res, statusCode.badRequest, message.requestNotOpen);
     }
+
+    const updatedRequest = await editOpenRequests(req.body, userId);
```

Comment:
Why not use `object destructuring` for the request body. And it seems your entry into the database is not validate. Please, do validate every thing that is being stored in our database.

Reply:
@chuksjoe .. Good observation

check the route to see that there is a validation attached.

the argument to the function has been destructured in the **editOpenRequests function expression**